connect/config.go

```diff
@@ -0,0 +1,99 @@
+package connect
+
+import (
+	"fmt"
+	"net"
+	"net/url"
+	"regexp"
+
+	"google.golang.org/grpc"
+	"google.golang.org/grpc/credentials/insecure"
+)
+
+type Config struct {
+	host string
+	port string
+	// requireTLS is whether TLS should be used or cleartext
+	requireTLS bool
+	// opts are additional options to pass to the gRPC client
+	opts []grpc.DialOption
+}
+
+// NewConfig returns a new Config from an endpoint string, that has to be in a URL format.
+// The endpoint can be either http:// or https:// that will be used to determine whether TLS should be used.
+// if port is not specified, it will be inferred from the scheme (80 for http, 443 for https).
+func NewConfig(endpoint string) (*Config, error) {
+	c := new(Config)
+	err := c.parseEndpoint(endpoint)
+	if err != nil {
+		return nil, fmt.Errorf("invalid endpoint: %w", err)
+	}
+	return c, nil
+}
+
+// GetConnectionURI returns connection string conforming to https://github.com/grpc/grpc/blob/master/doc/naming.md
+func (c *Config) GetConnectionURI() string {
+	return fmt.Sprintf("dns:%s:%s", c.host, c.port)
+}
+
+func (c *Config) RequireTLS() bool {
```

Comment:
(optional)

nit: if we have a `RequireTLS()` method and a `GetConnectionURI()` method, I'd expect to see a `GetOpts()` method too. Maybe remove these methods and let client.go use all the unexported Config fields directly? (up to you)

Reply:
tests live in `_tests` package, so it needs exported funcs

Reply:
It's not required to put your tests in a `_test` package, is it? I believe you can even have a mix of both, if you like.

From https://pkg.go.dev/testing:
> The test file can be in the same package as the one being tested, or in a corresponding package with the suffix "_test".
>
> If the test file is in the same package, it may refer to unexported identifiers within the package

Reply:
it is not really required, but I try to keep the following notation: 

- put each logical piece in separate `.go` file, 
- lowercase functions are implementation details, and not use them beyond this file if possible
- test only exported or _package private_ functions like above

Reply:
btw I added `GetDialOptions` for consistency